test/integration/partition_healing_test.js

```diff
@@ -0,0 +1,142 @@
+        ringpopA.membership.makeFaulty(addressB, initialIncarnationNumberB);
+        ringpopB.membership.makeFaulty(addressA, initialIncarnationNumberA);
+
+        ringpopA.healer.heal(function afterFirstHeal(err, targets) {
+            assert.ifError(err, 'healing successful');
+            assert.deepEqual(targets, [ringpopB.hostPort]);
+
+            assert.ok(ringpopA.membership.getIncarnationNumber() > initialIncarnationNumberA, 'node A reincarnated');
+            assert.ok(ringpopB.membership.getIncarnationNumber() > initialIncarnationNumberB, 'node B reincarnated');
+
+            ringpopA.healer.heal(function afterSecondHeal(err, targets) {
+                assert.ifError(err, 'healing successful');
+                assert.deepEqual(targets, [ringpopB.hostPort]);
+                assert.equal(ringpopA.membership.findMemberByAddress(addressB).status, 'alive', 'B is alive in A');
+                assert.equal(ringpopB.membership.findMemberByAddress(addressA).status, 'alive', 'A is alive in B');
+                assert.end();
+            });
+        });
+    }
+});
+
+function assertNoPartition(assert, cluster) {
+    _.each(cluster, function iterator(ringpop) {
+        _.each(ringpop.membership.members, assertAlive);
+    });
+
+    function assertAlive(member) {
+        assert.equal(member.status, 'alive');
+    }
+}
+
+testRingpopCluster({
+    size: 4,
+    waitForConvergence: true,
+    tapAfterConvergence: function tapAfterConvergence(cluster) {
+        GossipUtils.stopGossiping(cluster);
+    }
+}, 'healing - two partitions of two nodes', function t(bootRes, cluster, assert) {
+
+    GossipUtils.waitForNoGossip(cluster, test);
+
+    function test() {
+        var initialIncarnationNumbers = new Array(cluster.length);
+        for (var i = 0; i < cluster.length; i++) {
+            initialIncarnationNumbers[i] = cluster[i].membership.getIncarnationNumber();
+        }
+        var partitionA = [cluster[0], cluster[1]];
+        var partitionB = [cluster[2], cluster[3]];
+
+        _.each(partitionA, function(nodeA) {
+            _.each(partitionB, function(nodeB) {
+                nodeA.membership.makeFaulty(nodeB.hostPort, nodeB.membership.getIncarnationNumber());
+                nodeB.membership.makeFaulty(nodeA.hostPort, nodeA.membership.getIncarnationNumber());
+            });
+        });
+
+        for (var i = 0; i < partitionA.length; i++) {
+            var node = partitionA[i];
+            for (var j = 0; j < node.membership.members.length; j++) {
+                var member = node.membership.members[j];
+                if (_.pluck(partitionA, 'hostPort').indexOf(member.address) > -1) {
+                    assert.equal(member.status, 'alive')
+                } else if (_.pluck(partitionB, 'hostPort').indexOf(member.address) > -1) {
+                    assert.equal(member.status, 'faulty');
+                } else {
+                    assert.fail('!!!!');
```

Comment:
can we put a better description for the failure in the string? What is actually wrong when we reach this branch? It would make the output of the test more understandable as well as reading this test.